<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>451</width>
    <height>380</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Monal Log Viewer</string>
  </property>
  <property name="windowOpacity">
   <double>1.000000000000000</double>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>8</number>
    </property>
    <property name="topMargin">
     <number>10</number>
    </property>
    <property name="rightMargin">
     <number>10</number>
    </property>
    <item row="0" column="1">
     <widget class="QPushButton" name="uiButton_filterClear">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>&amp;Clear</string>
      </property>
     </widget>
    </item>
    <item row="0" column="0">
     <widget class="QComboBox" name="uiCombobox_filterInput">
      <property name="editable">
       <bool>true</bool>
      </property>
      <property name="insertPolicy">
       <enum>QComboBox::InsertAtBottom</enum>
      </property>
      <property name="duplicatesEnabled">
       <bool>false</bool>
      </property>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QSplitter" name="uiSplitter_inspectLine">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QListView" name="uiWidget_listView">
       <property name="verticalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="horizontalScrollMode">
        <enum>QAbstractItemView::ScrollPerPixel</enum>
       </property>
       <property name="layoutMode">
        <enum>QListView::SinglePass</enum>
       </property>
      </widget>
      <widget class="QTableWidget" name="uiTable_characteristics">
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QAbstractScrollArea::AdjustToContents</enum>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
       <property name="rowCount">
        <number>0</number>
       </property>
       <property name="columnCount">
        <number>2</number>
       </property>
       <attribute name="horizontalHeaderVisible">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderCascadingSectionResizes">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>80</number>
       </attribute>
       <attribute name="horizontalHeaderHighlightSections">
        <bool>true</bool>
       </attribute>
       <attribute name="horizontalHeaderShowSortIndicator" stdset="0">
        <bool>false</bool>
       </attribute>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderCascadingSectionResizes">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderMinimumSectionSize">
        <number>10</number>
       </attribute>
       <attribute name="verticalHeaderDefaultSectionSize">
        <number>800</number>
       </attribute>
       <attribute name="verticalHeaderHighlightSections">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderShowSortIndicator" stdset="0">
        <bool>false</bool>
       </attribute>
       <attribute name="verticalHeaderStretchLastSection">
        <bool>false</bool>
       </attribute>
       <column/>
       <column/>
      </widget>
     </widget>
    </item>
    <item row="2" column="0" colspan="2">
     <widget class="QFrame" name="uiFrame_search">
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>4</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>4</number>
       </property>
       <item>
        <widget class="QComboBox" name="uiCombobox_searchInput">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>250</width>
           <height>0</height>
          </size>
         </property>
         <property name="editable">
          <bool>true</bool>
         </property>
         <property name="insertPolicy">
          <enum>QComboBox::InsertAtBottom</enum>
         </property>
         <property name="duplicatesEnabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="uiButton_previous">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Previous</string>
         </property>
         <property name="shortcut">
          <string>Shift+F3</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="uiButton_next">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Next</string>
         </property>
         <property name="shortcut">
          <string>F3</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="QFrame" name="uiFrame_goToRow">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>4</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>4</number>
       </property>
       <item>
        <widget class="QSpinBox" name="uiSpinBox_goToRow">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="uiButton_goToRow">
         <property name="maximumSize">
          <size>
           <width>150</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Go to row</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="uiMenuBar_main">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>451</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="uiAction_open"/>
    <addaction name="uiAction_save"/>
    <addaction name="uiAction_export"/>
    <addaction name="separator"/>
    <addaction name="uiAction_close"/>
    <addaction name="uiAction_quit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="uiAction_preferences"/>
    <addaction name="uiAction_search"/>
    <addaction name="uiAction_inspectLine"/>
    <addaction name="separator"/>
    <addaction name="uiAction_pushStack"/>
    <addaction name="uiAction_popStack"/>
    <addaction name="separator"/>
    <addaction name="uiAction_goToRow"/>
    <addaction name="uiAction_firstRow"/>
    <addaction name="uiAction_lastRow"/>
    <addaction name="uiAction_firstRowInViewport"/>
    <addaction name="uiAction_lastRowInViewport"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="uiAction_about"/>
   </widget>
   <widget class="QMenu" name="uiMenu_profiles">
    <property name="title">
     <string>Profiles</string>
    </property>
    <addaction name="uiAction_deleteCurrentProfile"/>
    <addaction name="uiAction_cloneCurrentProfile"/>
    <addaction name="uiAction_addNewProfile"/>
    <addaction name="separator"/>
    <addaction name="uiAction_exportCurrentProfile"/>
    <addaction name="uiAction_importCurrentProfile"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="uiMenu_profiles"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="uiStatusbar_main"/>
  <action name="uiAction_open">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="statusTip">
    <string>Open rawlog file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="uiAction_quit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="statusTip">
    <string>Quit Whole LogViewer</string>
   </property>
  </action>
  <action name="uiAction_close">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="statusTip">
    <string>Close currently open file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="uiAction_preferences">
   <property name="text">
    <string>Preferences</string>
   </property>
   <property name="statusTip">
    <string>Open preferences</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="uiAction_search">
   <property name="text">
    <string>Search</string>
   </property>
   <property name="statusTip">
    <string>Search using python code</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="uiAction_export">
   <property name="text">
    <string>Export</string>
   </property>
   <property name="statusTip">
    <string>Export currently visible entries as text file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+E</string>
   </property>
  </action>
  <action name="uiAction_pushStack">
   <property name="text">
    <string>push Stack</string>
   </property>
   <property name="iconText">
    <string>push Stack</string>
   </property>
   <property name="statusTip">
    <string>Push current view (including filter, current search and scroll position) onto internal stack</string>
   </property>
   <property name="shortcut">
    <string>Ctrl++</string>
   </property>
  </action>
  <action name="uiAction_popStack">
   <property name="text">
    <string>pop Stack</string>
   </property>
   <property name="statusTip">
    <string>Pop last pushed view off internal stack</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="uiAction_save">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="statusTip">
    <string>Save currently visible entries as rawlog</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="uiAction_inspectLine">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show details</string>
   </property>
   <property name="statusTip">
    <string>Open of all structured logging details for the currently selected log entry</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+D</string>
   </property>
  </action>
  <action name="uiAction_about">
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="uiAction_goToRow">
   <property name="text">
    <string>Go to row</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="uiAction_firstRow">
   <property name="text">
    <string>First row</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Home</string>
   </property>
  </action>
  <action name="uiAction_lastRow">
   <property name="text">
    <string>Last row</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+End</string>
   </property>
  </action>
  <action name="uiAction_firstRowInViewport">
   <property name="text">
    <string>First row in viewport</string>
   </property>
   <property name="shortcut">
    <string>Home</string>
   </property>
  </action>
  <action name="uiAction_lastRowInViewport">
   <property name="text">
    <string>Last row in viewport</string>
   </property>
   <property name="shortcut">
    <string>End</string>
   </property>
  </action>
  <action name="uiAction_cloneCurrentProfile">
   <property name="text">
    <string>Clone Current Profile</string>
   </property>
  </action>
  <action name="uiAction_addNewProfile">
   <property name="text">
    <string>Add New Profile</string>
   </property>
  </action>
  <action name="uiAction_deleteCurrentProfile">
   <property name="text">
    <string>Delete Current Profile</string>
   </property>
  </action>
  <action name="uiAction_exportCurrentProfile">
   <property name="text">
    <string>Export Current Profile</string>
   </property>
  </action>
  <action name="uiAction_importCurrentProfile">
   <property name="text">
    <string>Import New Profile</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
